kogito-codegen-modules/kogito-codegen-core/src/main/java/org/kie/kogito/codegen/core/CodegenUtils.java

```diff
@@ -81,6 +84,23 @@ public static void interpolateArguments(MethodDeclaration md, String dataType) {
         md.getParameters().forEach(p -> p.setType(dataType));
     }
 
+    public static void interpolateArguments(NodeWithArguments<? extends Node> node, Map<String, Expression> expressions) {
+        List<Expression> argumentList = node.getArguments();
+        Map<String, Integer> argumentIds = new HashMap<>();
+
+        expressions.forEach((placeholder, expr) -> {
+            for (int i = 0; i < argumentList.size(); i++) {
+                if (argumentList.get(i).toString().contains(placeholder)) {
```

Comment:
Why `contains`? What about
```suggestion
                if (argumentList.get(i).toString().trim().equals(placeholder)) {
```